TypeCobol.Analysis.Test/BasicCfgInstrs/PerformGoesOutOfbounds0.dot

```diff
@@ -0,0 +1,36 @@
+﻿digraph Cfg {
+compound=true;
+node [
+shape = "record"
+]
+
+edge [
+arrowtail = "empty"
+]
+Block0 [
+label = "{START|}"
+]
+Block1 [
+label = "{Block1|    perform a\l}"
+]
+subgraph cluster_1{
+color = blue;
+label = "a";
+Block4 [
+label = "{A. Block4|    display 'a'\l    go to b\l}"
+]
+Block6 [
+label = "{B. Block6|    display 'b'\l}"
+]
+Block4 -> Block6
+
+}
+Block1 -> Block4 [style=dashed]
+
+Block2 [
+label = "{Block2|    goback\l}"
+]
+Block0 -> Block1
+Block1 -> Block2
```

Comment:
The code contains an error:
`Line 4[12,12] <30, Error, Semantics> - Semantic error: Statement '    go to b' located at line 10, column 12 prevents this PERFORM statement to reach its exit.`

IBM Compiler only reports a warning, because the code can still be executed.
Execution will produce an abend U4038, because there  is no goback at the end of paragraph `b`. But if there is a goback, the execution is ok.
So shall we create an error or a warning? I think it should be a warning to behave like IBM compiler.
We can then ask our users if they want an error. In this case the error could be set as parameter of CLI/LSP (like parameter `diagnostic.checkEndAlignment`)


For the CFG : the goback statement can never be reached, but the dot file shows otherwise.
Line Block1 -> Block2 should not appear in the dot file.

Reply:
Ok to turn this diagnostic into a Warning.

About the Block1 to Block2 edge, this edge is created while building the graph. It is only after resolving all GOTO/ALTER/PERFORM that we could find out which part of the code is unreachable or not.
The graph is built as to contain all and every possible paths but we need a post-build analysis to actually find which parts are unreachable. Do you confirm @mayanje ?

Reply:
In my opinion this case the error is that the perfom goes out of paragraph a and thus the goback will never be executed, but this a consequence of the **abend U4038** and not the execution flow. So you are right to stay iso with IBM Compiler it is a warging (a Severe Warning).
The unreachability of the goback instruction can be detected by an abstract interpretation in this case, because we must detect that finally we will face an **abend U4038**.